File: take_u_forward/recursion/2_ Problems on Recursion.cpp
//Q1. Print name N times using Recursion

#include <iostream>

void fun(int i,int n){
    if(i>n){
        return;
    }
    std::cout<<"Saifuddin\n";
    fun(i+1,n);
}

int main()
{
    int n;
    std::cout<<"Enter the value of N:\t";
    std::cin>>n;
    
    fun(1,n);

    return 0;
}

//Q2. Print 1 to N

#include <iostream>

void fun(int i,int n){
    if(i>n){
        return;
    }
    std::cout<<i<<"\n";
    fun(i+1,n);
}

int main()
{
    int n;
    std::cout<<"Enter the value of N:\t";
    std::cin>>n;
    
    fun(1,n);

    return 0;
}

File: take_u_forward/recursion/2_ Problems on Recursion.java
// Print name N times using Recursion

import java.util.Scanner;
public class Main
{
    static void fun(int i,int n){
        if(i>n){
            return;
        }
        System.out.println("Saifuddin");
        fun(i+1,n);
    }
	public static void main(String[] args) {
		
		Scanner sc=new Scanner(System.in);
		
		System.out.print("Enter the the value of N: ");
		int n=sc.nextInt();
		
		fun(1,n);
		
	}
}


//Q2. Print 1 to N

import java.util.Scanner;
public class Main
{
    static void fun(int i,int n){
        if(i>n){
            return;
        }
        System.out.println(i);
        fun(i+1,n);
    }
	public static void main(String[] args) {
		
		Scanner sc=new Scanner(System.in);
		
		System.out.print("Enter the the value of N: ");
		int n=sc.nextInt();
		
		fun(1,n);
		
	}
}